### ISSUE TITLE
409 Conflict beim Anlegen eines Interessenten (POST /Prospects

### ISSUE THREAD

**Issue by timothy3001** [selectline-software/selectline-api#377]
Bei einem unserer Kunden setzen wir die SelectLine-REST-API ein, um automatisiert Interessenten in SelectLine anzulegen. Dazu nutzen wir folgenden SelectLine-REST-API-Endpunkt:

`POST /Prospects`

In manchen Fällen (sehr unregelmäßig) bekommen wir einen 409 Conflict als HTTP Antwort. In der Doku konnte ich nichts zu dem Status-Code finden und wir können diesen Fehler nicht provozieren.

Unsere Vermutung ist, dass es evtl. mit der Auslastung von SelectLine zu tun hat und Ressourcen zufällig blockiert sind. Wir haben bereits einen "Retry"-Mechanismus hinzugefügt, der aber nicht immer korrekt greift.

Da der Fehler nicht dokumentiert ist, könnten Sie uns vielleicht mitteilen, was mögliche Ursachen für den 409 beim POST eines Interessenten sein könnten?

### Details
- Current version number: 23.2.3
- [x] Test status checked in Mobile Manager
- Function or route (request): POST /Prospects
- Error (response): 409 Conflict

#### Steps to replicate the behavior:
unbekannt

**Comment by MatthiasGuse** [comment]
Hallo,

was schicken Sie beispielhaft im Body mit?

In der Regel steht die Ursache in der Message des Response-Bodys. Bspw. "Prospect 'xyz' existiert bereits."

Grüße

**Comment by timothy3001** [comment]
Ok, vielen dank schonmal. Im Body beim Senden des Interessenten stehen eigentlich gültige Daten, insbesondere Duplikate prüfen wir vorher, da der Kunde dies tatsächlich auch vorab schon gewünscht hat.

Ich füge unserem Code Trace-Logging hinzu und melde mich, sobald der Fehler wieder aufgetreten ist.

**Comment by timothy3001** [comment]
Wir haben das Logging erweitert und konnten den Fehler mittlerweile nachstellen.

Ich kann leider gerade nicht auf das System zugreifen, sodass ich die exakte Fehlermeldung nicht habe (reiche ich nach). Aber: Es handelt sich bei uns auch um die "EntityAlreadyExistsException" Problematik.

Im Unterschied zu @Micha-Richter senden wir aber nur einen Request und erhalten den Fehler bereits.

Meine Vermutung: In SelectLine gibt es mittlerweile eine eigene Duplikatsprüfung für die Interessenten? Bzw. für die Tabelle CRM_Adressen. 

@Micha-Richter Bei Ihnen sind beide CURL-Befehle identisch, wenn ich das richtig sehe. Haben Sie mal versucht, eine Adresse abzuwandeln?

@MatthiasGuse Können Sie evtl. etwas Licht ins Dunkel bringen?

**Comment by Micha-Richter** [comment]
> @Micha-Richter Bei Ihnen sind beide CURL-Befehle identisch, wenn ich das richtig sehe. Haben Sie mal versucht, eine Adresse abzuwandeln?

Ja, das ändert aber nichts (gerade noch einmal geprüft). 

Und die Message im Ergebnis `"Prospect '13' existiert bereits."` deutet auch auf den SelectLine-Schlüssel in der Tabelle als Ursache, die eigentlichen Nutzerdaten spielen hier keine Rolle (ich kann zB immer wieder einen der beiden Datensätze anlegen, s. Screenshot):

<img width="757" alt="image" src="https://github.com/selectline-software/selectline-api/assets/12509902/6a50e47e-f23b-4345-8ea1-d7807d3e2824">

**Comment by timothy3001** [comment]
Stimmt, habe nochmal ein weiteres Beispiel gesucht, dort erkennt unsere Automatisierung kein Duplikat.

Wie versprochen, hier bei uns der Response Body beim POST auf /prospects:

```
{
  "StatusCode": "Conflict",
  "ResponseCode": "12-009",
  "ResponseId": 9,
  "ResponseIdDescription": "EntityAlreadyExistsException",
  "Message": "Prospect '1093326' existiert bereits.",
  "Details": null
}
```

**Comment by MatthiasGuse** [comment]
Hallo,

sofern niemand die Datensatzschlüssel manipuliert, kann das eigentlich nur durch gleichzeitige Zugriffe vorkommen.
Ich werde für einen Workaround ein Workitem bei uns erfassen. Hier muss die API toleranter werden und mit solchen Fällen umgehen können. Einen zeitlichen Horizont kann ich allerdings aktuell noch nicht nennen.

Viele Grüße
